Example/ScanDemo 2/Pods/CML_SCAN/ScanView.swift

```diff
@@ -0,0 +1,284 @@
+//
+//  ScanView.swift
+//  SCAN
+//
+//  Created by sukhjeet singh sandhu on 27/07/17.
+//  Copyright © 2017 Solcen. All rights reserved.
+//
+
+import UIKit
+import AVFoundation
+import AudioToolbox
+
+// These are the barcode types, those are supported by this Pod.
+public enum Types {
+    case qrCode
+    case dataMatrix
+    case ean8
+    case ean13
+    case code128
+    case code39
+}
+
+// This protocol is responsibe for providing the barcode and type back to the view controller.
+public protocol BarcodeScannerDelegate: class {
+    func didReceiveBarcode(barcode: String?, barcodeType: String?)
+}
+
+open class ScanView: UIView {
+
+    var isBackCamera = true
+
+    fileprivate var session: AVCaptureSession!
+    fileprivate var previewLayer: AVCaptureVideoPreviewLayer!
+    fileprivate var interestRect: CGRect!
+    fileprivate var error: String?
+
+    var supportedTypes: [String] = []
+    var shouldVibrateOnScan = true
+    var topMask: UIView!
+    var bottomMask: UIView!
+    var leftMask: UIView!
+    var rightMask: UIView!
+    var topLHLine: UIView!
+    var topRHLine: UIView!
+    var bottomLHLine: UIView!
+    var bottomRHLine: UIView!
+    var topLVLine: UIView!
+    var topRVLine: UIView!
+    var bottomLVLine: UIView!
+    var bottomRVLine: UIView!
+    var currentOrientation: UIDeviceOrientation?
+    var captureDevice: AVCaptureDevice?
+
+    open weak var barcodeDelegate: BarcodeScannerDelegate?
+
+    // This function should be called, in order to start scanning with supported barcode types as parameter.
+    open func startScanning(with types: [Types], onError: ((String?)->())) {
+        NotificationCenter.default.addObserver(self, selector: #selector(ScanView.handleRotation), name: NSNotification.Name.UIApplicationDidChangeStatusBarOrientation, object: nil)
+        self.backgroundColor = .white
+        self.setSupportesTypes(from: types)
+        self.loadSubViews()
+        self.scan()
+        if let error = self.error {
+            onError(error)
+            self.error = nil
+        } else {
+            onError(nil)
+        }
+    }
+
+    // Call this function to switch the camera from back to front or vice-versa.
+    open func switchCamera() {
```

Comment:
Are you sure you understand the difference between `open` and `public`? Why would you want this to be overridden?